In [1]:
import pandas as pd 
pd.set_option('display.max_columns', None)
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import warnings
warnings.filterwarnings('ignore')



<h1> 1. Upload and take a first look to the dataset <h1>

<h4> 1.1. Upload <h4>

In [2]:
dataset = pd.read_csv('Speed+Dating+Data.csv', encoding = 'ISO-8859-1')

In [3]:
dataset.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   
3        4  14.0      1      0.61         0   23.0     2.0      30.0   
4        5  15.0      1      0.21         0   24.0     3.0      30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   
3       5.0      15.0      40.0       5.0       5.0      1     7.0     8.0   
4      10.0      20.0      10.0      10.0      20.0      1     8.0     7.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
3  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
4  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
3       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
4       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
3      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
4      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
3     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
4     15.0     

<h4> 1.2. Overview and basic stats <h4>

In [5]:
print(f"Number of rows : {dataset.shape[0]}")
print(f"Number of columns : {dataset.shape[1]}")
print()

print("Display of dataset: ")
display(dataset.head(10))
print()

print("Basics statistics: ")
display(dataset.describe(include='all'))
print()
pd.options.display.max_rows = 999
print("Percentage of missing values: ")
display((100 * dataset.isnull().sum() /dataset.shape[0]).sort_values(ascending= False).reset_index())

Number of rows : 8378
Number of columns : 195

Display of dataset: 


iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   
5    1  1.0       0    1       1     1     10         7       NaN      6   
6    1  1.0       0    1       1     1     10         7       NaN      1   
7    1  1.0       0    1       1     1     10         7       NaN      2   
8    1  1.0       0    1       1     1     10         7       NaN      8   
9    1  1.0       0    1       1     1     10         7       NaN      9   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0     35.00   
1        2  12.0      0      0.54         0   22.0     2.0     60.00   
2        3  13.0      1      0.16         1   22.0     4.0     19.00   
3        4  14.0      1      0.61         0   23.0     2.0     30.00   
4        5  15.0      1      0.21         0   24.0     3.0     30.00   
5        6  16.0      0      0.25         0   25.0     2.0     50.00   
6        7  17.0      0      0.34         0   30.0     2.0     35.00   
7        8  18.0      0      0.50         0   27.0     2.0     33.33   
8        9  19.0      1      0.28         0   28.0     2.0     50.00   
9       10  20.0      0     -0.36         0   24.0     2.0    100.00   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0     20.00     20.00     20.00      0.00      5.00      0     6.0     8.0   
1      0.00      0.00     40.00      0.00      0.00      0     7.0     8.0   
2     18.00     19.00     18.00     14.00     12.00      1    10.0    10.0   
3      5.00     15.00     40.00      5.00      5.00      1     7.0     8.0   
4     10.00     20.00     10.00     10.00     20.00      1     8.0     7.0   
5      0.00     30.00     10.00      0.00     10.00      1     7.0     7.0   
6     15.00     25.00     10.00      5.00     10.00      0     3.0     6.0   
7     11.11     11.11     11.11     11.11     22.22      0     6.0     7.0   
8      0.00     25.00     10.00      0.00     15.00      1     7.0     7.0   
9      0.00      0.00      0.00      0.00      0.00      0     6.0     6.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   
5      8.0    8.0    7.0     7.0     7.0     6.0    2.0  21.0   Law       1.0   
6      7.0    5.0    8.0     7.0     2.0     1.0    2.0  21.0   Law       1.0   
7      5.0    6.0    8.0     6.0     7.0     5.0    2.0  21.0   Law       1.0   
8      8.0    8.0    8.0     9.0     6.5     8.0    2.0  21.0   Law       1.0   
9      6.0    6.0    6.0     6.0     6.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
5      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
6      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
7      NaN    NaN     NaN   4.0      2.0       4.0  Ch


Basics statistics: 


iid           id       gender          idg       condtn  \
count   8378.000000  8377.000000  8378.000000  8378.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean     283.675937     8.960248     0.500597    17.327166     1.828837   
std      158.583367     5.491329     0.500029    10.940735     0.376673   
min        1.000000     1.000000     0.000000     1.000000     1.000000   
25%      154.000000     4.000000     0.000000     8.000000     2.000000   
50%      281.000000     8.000000     1.000000    16.000000     2.000000   
75%      407.000000    13.000000     1.000000    26.000000     2.000000   
max      552.000000    22.000000     1.000000    44.000000     2.000000   

               wave        round     position     positin1        order  \
count   8378.000000  8378.000000  8378.000000  6532.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      11.350919    16.872046     9.042731     9.295775     8.927668   
std        5.995903     4.358458     5.514939     5.650199     5.477009   
min        1.000000     5.000000     1.000000     1.000000     1.000000   
25%        7.000000    14.000000     4.000000     4.000000     4.000000   
50%       11.000000    18.000000     8.000000     9.000000     8.000000   
75%       15.000000    20.000000    13.000000    14.000000    13.000000   
max       21.000000    22.000000    22.000000    22.000000    22.000000   

            partner          pid        match     int_corr     samerace  \
count   8378.000000  8368.000000  8378.000000  8220.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       8.963595   283.863767     0.164717     0.196010     0.395799   
std        5.491068   158.584899     0.370947     0.303539     0.489051   
min        1.000000     1.000000     0.000000    -0.830000     0.000000   
25%        4.000000   154.000000     0.000000    -0.020000     0.000000   
50%        8.000000   281.000000     0.000000     0.210000     0.000000   
75%       13.000000   408.000000     0.000000     0.430000     1.000000   
max       22.000000   552.000000     1.000000     0.910000     1.000000   

              age_o       race_o     pf_o_att     pf_o_sin     pf_o_int  \
count   8274.000000  8305.000000  8289.000000  8289.000000  8289.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      26.364999     2.756653    22.495347    17.396867    20.270759   
std        3.563648     1.230689    12.569802     7.044003     6.782895   
min       18.000000     1.000000     0.000000     0.000000     0.000000   
25%       24.000000     2.000000    15.000000    15.000000    17.390000   
50%       26.000000     2.000000    20.000000    18.370000    20.000000   
75%       28.000000     4.000000    25.000000    20.000000    23.810000   
max       55.000000     6.000000   100.000000    60.000000    50.000000   

           pf_o_fun     pf_o_amb     pf_o_sha        dec_o       attr_o  \
count   8280.000000  8271.000000  8249.000000  8378.000000  8166.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      17.459714    10.685375    1


Percentage of missing values: 


index          0
0    num_in_3  92.026737
1    numdat_3  82.143710
2      expnum  78.515159
3     sinc7_2  76.665075
4      amb7_2  76.665075
5     shar7_2  76.438291
6      fun7_2  76.318931
7     attr7_2  76.318931
8    intel7_2  76.318931
9     attr5_3  75.936978
10     fun7_3  75.936978
11     fun5_3  75.936978
12    attr7_3  75.936978
13     amb7_3  75.936978
14     amb5_3  75.936978
15   intel5_3  75.936978
16   intel7_3  75.936978
17    shar7_3  75.936978
18    sinc7_3  75.936978
19    sinc5_3  75.936978
20    shar2_3  75.936978
21    shar4_3  64.681308
22     amb4_3  64.681308
23    attr2_3  64.681308
24   intel2_3  64.681308
25    sinc2_3  64.681308
26     amb2_3  64.681308
27     fun2_3  64.681308
28    attr4_3  64.681308
29    sinc4_3  64.681308
30   intel4_3  64.681308
31     fun4_3  64.681308
32     mn_sat  62.604440
33    tuition  57.233230
34    attr3_3  52.566245
35   them_cal  52.566245
36   you_call  52.566245
37    attr1_3  52.566245
38     date_3  52.566245
39     amb1_3  52.566245
40    shar1_3  52.566245
41    sinc3_3  52.566245
42   intel3_3  52.566245
43     amb3_3  52.566245
44    sinc1_3  52.566245
45     fun1_3  52.566245
46     fun3_3  52.566245
47   intel1_3  52.566245
48     amb3_s  52.255908
49    attr3_s  52.255908
50    sinc3_s  52.255908
51     fun3_s  52.255908
52   intel3_s  52.255908
53   intel1_s  51.110050
54     fun1_s  51.110050
55    shar1_s  51.110050
56    attr1_s  51.110050
57    sinc1_s  51.110050
58     amb1_s  51.110050
59     income  48.925758
60    sinc5_2  47.756028
61   intel5_2  47.756028
62     fun5_2  47.756028
63     amb5_2  47.756028
64    attr5_2  47.756028
65     fun5_1  41.441872
66   intel5_1  41.441872
67    attr5_1  41.441872
68     amb5_1  41.441872
69    sinc5_1  41.441872
70   undergra  41.346383
71   intel4_2  31.069468
72     fun4_2  31.069468
73    attr4_2  31.069468
74    shar2_2  31.069468
75     amb4_2  31.069468
76    sinc4_2  31.069468
77     fun2_2  31.069468
78     amb2_2  31.069468
79   intel2_2  31.069468
80    sinc2_2  31.069468
81    shar4_2  31.069468
82    attr2_2  31.069468
83    shar4_1  22.809740
84    attr4_1  22.547147
85     amb4_1  22.547147
86    sinc4_1  22.547147
87   intel4_1  22.547147
88     fun4_1  22.547147
89   positin1  22.033898
90   match_es  14.000955
91     shar_o  12.843161
92       shar  12.735736
93    zipcode  12.699928
94   numdat_2  11.279542
95    attr1_2  11.136309
96   intel1_2  10.921461
97    shar1_2  10.921461
98     amb1_2  10.921461
99     fun1_2  10.921461
100   sinc1_2  10.921461
101   satis_2  10.921461
102    amb3_2  10.921461
103    fun3_2  10.921461
104  intel3_2  10.921461
105   sinc3_2  10.921461
106   attr3_2  10.921461
107    length  10.921461
108     amb_o   8.617809
109       amb   8.498448
110     met_o   4.595369
111       met   4.476009
112     fun_o   4.296968
113       fun   4.177608
114    prob_o   3.795655
115      prob   3.688231
116   intel_o   3.652423
117     intel   3.533063
118    sinc_o   3.425639
119      sinc   3.306278
120    like_o   2.984006
121      like   2.864646
122    attr_o   2.530437
123      attr   2.411077
124  int_corr   1.885892
125  career_c   1.647171
126  pf_o_sha   1.539747
127   shar1_1   1.444259
128  pf_o_amb   1.277154
129  intel3_1   1.253282
130    fun3_1   1.253282
131    amb3_1   1.253282
132   sinc3_1   1.253282
133   attr3_1   1.253282
134     age_o   1.241346
135  exphappy   1.205538
136    amb1_1   1.181666
137  pf_o_fun   1.169730
138      date   1.157794
139       age   1.133922
140    fun1_1   1.062306
141  pf_o_sin   1.062306
142  pf_o_att   1.062306
143    career   1.062306
144   shar2_1   1.062306
145    amb2_1   1.062306
146  pf_o_int   1.062306
147  field_cd   0.978754
148  concerts   0.942946
149     music   0.942946
150   theater   0.942946
151  shopping   0.942946
152       art   0.942946
153    hiking   0.942946
154    gaming   0.942946
155    movies   0.942946
156  clubbing   0.942946
157        tv   0.942946
158   reading   0.942946
159   si

We can already see that there is a lot of columnns, a lot of missing values, but not that much of rows (8378). Therefore in order to work more properly we will have to carefully select the columns that we are gonna use for the analysis, and perform some cleaning.

<h1> 2. Cleaning of the dataset <h1>

<h4> 2.1. Remove column with more than 50% missing values <h4>

In [6]:
def too_much_missing_drop(df):
    for col in [*df.columns]:
        if (100 * df.loc[:,col].isnull().sum()/df.shape[0]) > 50:
            df = df.drop(col, axis = 1)
            print(df.shape)
    return df
 
dataset = too_much_missing_drop(dataset)
dataset.shape

(8378, 194)
(8378, 193)
(8378, 192)
(8378, 191)
(8378, 190)
(8378, 189)
(8378, 188)
(8378, 187)
(8378, 186)
(8378, 185)
(8378, 184)
(8378, 183)
(8378, 182)
(8378, 181)
(8378, 180)
(8378, 179)
(8378, 178)
(8378, 177)
(8378, 176)
(8378, 175)
(8378, 174)
(8378, 173)
(8378, 172)
(8378, 171)
(8378, 170)
(8378, 169)
(8378, 168)
(8378, 167)
(8378, 166)
(8378, 165)
(8378, 164)
(8378, 163)
(8378, 162)
(8378, 161)
(8378, 160)
(8378, 159)
(8378, 158)
(8378, 157)
(8378, 156)
(8378, 155)
(8378, 154)
(8378, 153)
(8378, 152)
(8378, 151)
(8378, 150)
(8378, 149)
(8378, 148)
(8378, 147)
(8378, 146)
(8378, 145)
(8378, 144)
(8378, 143)
(8378, 142)
(8378, 141)
(8378, 140)
(8378, 139)
(8378, 138)
(8378, 137)
(8378, 136)


(8378, 136)

<h4> 2.2. Replacing the rest of missing values by the mean for each column that we are gonna use<h4>

In [7]:
col_to_fill = ['attr', 'sinc', 'intel', 'fun', 'amb', 'shar','attr3_1','sinc3_1', 'intel3_1', 'fun3_1', 'amb3_1']

for col in col_to_fill:
    dataset.loc[:,col] = dataset.loc[:,col].fillna(dataset.loc[:,col].mean())
    dataset.loc[:,col] = dataset.loc[:,col].apply(lambda x : round(x))

<h4>2.3. Copy of the original dataset, to keep it clean from our preprocessing/exploration work <h4>

In [8]:
dataset_explo = dataset.copy()

<h4>2.4. Making the data a bit more understandable <h4>

Rename of some columns for a better understanding of the dataset

In [9]:

dataset_explo = dataset_explo.rename(columns={"iid": "candidate_id",
                                            "condtn": "choice",
                                            "goal": "goal",
                                            "round": "candidates_in_wave",
                                            "pid":"partner_id",
                                            "int_corr":"canidates_intrests_corr",
                                            "age_o":"age_partner",
                                            "age":"candidate_age",
                                            "imprelig":"religion_importance",
                                            "date":"date_frequence",
                                            "go_out":"go_out_frequence",
                                            "career_c":"career_candidate",
                                            "dec":"decision",
                                            "pf_o_att" : "partner_pref_att",
                                            "dec_o" : "decision_partner_night",
                                            "attr_o" : "rating_attrac_by_part",
                                            "sinc_o" : "rating_sinc_by_part",
                                            "intel_o" : "rating_intel_by_part",
                                            "fun_o" : "rating_fun_by_part",
                                            "amb_o" : "rating_amb_by_part",
                                            "shar_o" : "rating_shar_intrest_by_part",
                                            "numdat_2":"nb_dates",
                                            "attr1_1":"look_for_attr",
                                            "sinc1_1":"look_for_sincere",
                                            "intel1_1":"look_for_intel",
                                            "fun1_1":"look_for_fun",
                                            "amb1_1":"look_for_amb",
                                            "shar1_1":"look_for_sharedint",        
                                            "attr3_1":"own_attr_attr",
                                            "sinc3_1":"own_attr_sincere",
                                            "intel3_1":"own_attr_intel",
                                            "fun3_1":"own_attributes_fun",
                                            "amb3_1":"own_attr_amb",
                                            "shar3_1":"own_attr_sharedint"                                     
                                           })



Replacing code by their meaning

In [10]:
#goal code
dataset_explo['goal'] = dataset_explo['goal'].apply(lambda x: 'Seemed like a fun night out' if x == 1 
                                                else 'To meet new people' if x == 2 
                                                else 'To get a date' if x == 3
                                                else 'Looking for a serious relationship' if x == 4
                                                else 'To say I did it' if x == 5
                                                else 'Other' if x == 6
                                                else  x
                                                )
#gender core
dataset_explo['gender'] = dataset_explo['gender'].apply(lambda x: 'Female' if x == 0 else 'Male' if x ==1 else x)

# Match code
dataset_explo['match'] = dataset_explo['match'].apply(lambda x: 'No' if x == 0 else 'Yes')

# Choice given per wave
dataset_explo['choice'] = dataset_explo['choice'].apply(lambda x: 'limited choice' if x == 1 else 'extensive choice')

In [11]:
dataset_explo.head()

candidate_id   id  gender  idg          choice  wave  candidates_in_wave  \
0             1  1.0  Female    1  limited choice     1                  10   
1             1  1.0  Female    1  limited choice     1                  10   
2             1  1.0  Female    1  limited choice     1                  10   
3             1  1.0  Female    1  limited choice     1                  10   
4             1  1.0  Female    1  limited choice     1                  10   

   position  positin1  order  partner  partner_id match  \
0         7       NaN      4        1        11.0    No   
1         7       NaN      3        2        12.0    No   
2         7       NaN     10        3        13.0   Yes   
3         7       NaN      5        4        14.0   Yes   
4         7       NaN      7        5        15.0   Yes   

   canidates_intrests_corr  samerace  age_partner  race_o  partner_pref_att  \
0                     0.14         0         27.0     2.0              35.0   
1                     0.54         0         22.0     2.0              60.0   
2                     0.16         1         22.0     4.0              19.0   
3                     0.61         0         23.0     2.0              30.0   
4                     0.21         0         24.0     3.0              30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  decision_partner_night  \
0      20.0      20.0      20.0       0.0       5.0                       0   
1       0.0       0.0      40.0       0.0       0.0                       0   
2      18.0      19.0      18.0      14.0      12.0                       1   
3       5.0      15.0      40.0       5.0       5.0                       1   
4      10.0      20.0      10.0      10.0      20.0                       1   

   rating_attrac_by_part  rating_sinc_by_part  rating_intel_by_part  \
0                    6.0                  8.0                   8.0   
1                    7.0                  8.0                  10.0   
2                   10.0                 10.0                  10.0   
3                    7.0                  8.0                   9.0   
4                    8.0                  7.0                   9.0   

   rating_fun_by_part  rating_amb_by_part  rating_shar_intrest_by_part  \
0                 8.0                 8.0                          6.0   
1                 7.0                 7.0                          5.0   
2                10.0                10.0                         10.0   
3                 8.0                 9.0                          8.0   
4                 6.0                 9.0                          7.0   

   like_o  prob_o  met_o  candidate_age field  field_cd undergra  race  \
0     7.0     4.0    2.0           21.0   Law       1.0      NaN   4.0   
1     8.0     4.0    2.0           21.0   Law       1.0      NaN   4.0   
2    10.0    10.0    1.0           21.0   Law       1.0      NaN   4.0   
3     7.0     7.0    2.0           21.0   Law       1.0      NaN   4.0   
4     8.0     6.0    2.0           21.0   Law       1.0      NaN   4.0   

   imprace  religion_importance     from zipcode     income  \
0      2.0                  4.0  Chicago  60,521  69,487.00   
1      2.0                  4.0  Chicago  60,521  69,487.00   
2      2.0                  4.0  Chicago  60,521  69,487.00   
3      2.0                  4.0  Chicago  60,521  69,487.00   
4      2.0                  4.0  Chicago  60,521  69,487.00   

                 goal  date_frequence  go_out_frequence  career  \
0  To meet new people             7.0               1.0  lawyer   
1  To meet new people             7.0               1.0  lawyer   
2  To meet new people             7.0               1.0  lawyer   
3  To meet new people             7.0               1.0  lawyer   
4  To meet new people             7.0               1.0  lawyer   

   career_candidate  sports  tvsports  exercise  dining  museums  art  hiking  \
0               NaN     9.0       2.0      

<h1> 3. Data Exploration  <h1>

Now that the dataset is a bit more clean and understandable, we will be able to start exploring to have a better knowledge of it and maybe get some hint on which analysis perform and how to answer the question.

Creation of a dataset without duplicates id in order to avoid counting the same people multiple time during some of the exploration.

In [12]:
dataset_no_doublons = dataset_explo.drop_duplicates('candidate_id')

<h4> 3.1. Looking at the profil of the participant <h4>

In [13]:
fig_gender_repart = px.histogram(dataset_no_doublons, x = 'gender', color = 'gender', title = 'Gender repartition', template="simple_white", width= 600)
fig_gender_repart.update(layout=dict(title=dict(x=0.5)))
fig_gender_repart.show()

The dataset is equilibrate in gender, which is pretty good as a desequilibrate 'dating market' could create bias.

In [14]:
fig_age_repart = px.histogram(dataset_no_doublons, x = 'candidate_age', color = 'gender', title = 'Age distribution by gender', width= 600, template="simple_white", labels = {'candidate_age' : 'Age' })
fig_age_repart.update(layout=dict(title=dict(x=0.5)))
fig_age_repart.show()


The age distribution revolve between 20 and 30, the participants are relatively young. This can guide us on their match decision.

In [15]:
fig = px.pie( values=dataset_explo['goal'].value_counts().values, names=dataset_explo['goal'].value_counts().index, title='Goals of the speed dating for the participants', width = 700, height = 500)
fig.update(layout=dict(title=dict(x=0.5)))
fig.update_traces(pull=[0, 0, 0, 0,0,0.1])
fig.show()

The goals of the speed dating are superficial for most of the participants ('have fun', 'get a date', 'to say I did it'), with only 3.63% looking for a serious relation.
We can assume that this information will affect the reasons of the match, maybe based on also more superficial reasons, like the physical appearence.

In [16]:
dataset_male = dataset_explo.loc[dataset_explo['gender'] == 'Male',:]
dataset_female = dataset_explo.loc[dataset_explo['gender'] == 'Female',:]

categories = ['Sincerity', 'Intelligence', 'Fun', 'Ambitious', 'Attractive', 'Shared interest']

fig_looking_for = go.Figure()

fig_looking_for.add_trace(go.Scatterpolar(
      r=[dataset_male['look_for_sincere'].mean(), dataset_male['look_for_intel'].mean() , dataset_male['look_for_fun'].mean(),dataset_male['look_for_amb'].mean(),dataset_male['look_for_attr'].mean(),dataset_male['look_for_sharedint'].mean() ],
      theta=categories,
      fill='toself',
      name='Male'
))
fig_looking_for.add_trace(go.Scatterpolar(
      r=[dataset_female['look_for_sincere'].mean(), dataset_female['look_for_intel'].mean() , dataset_female['look_for_fun'].mean(),dataset_female['look_for_amb'].mean(),dataset_female['look_for_attr'].mean(), dataset_male['look_for_sharedint'].mean()],
      theta=categories,
      fill='toself',
      name='Female'
))

fig_looking_for.update_layout(
  polar=dict(
    radialaxis=dict(
      visible=True,
      range=[0, 30]
    )),
  showlegend= True,
  title = go.layout.Title(text = "Mean declared importance of features for participants by gender"
                          , x = 0.5),
  width=700,
  height=700
)

fig_looking_for.show()

As previously assumed we can see that the physical attractivity is the feature declared as the most important, especially for men. For women other features seems to be as important like the sincerity, and even 
slightly more important like the intelligence. A declared importance for more 'deep'/personality based features like the sincerity contrast with the previously declared goals, it could be intresting to check how it is in reality.

<h4> 3.2. Looking at the match percentage <h4>

In [17]:
fig_match_perc = px.histogram(x = dataset_explo.loc[:, 'match'].unique(), y =[((len(dataset_explo.loc[dataset_explo['match'] == 'No', :]))/len(dataset_explo))*100, (len(dataset_explo.loc[dataset_explo['match'] == 'Yes', :]))/len(dataset_explo)*100], title = 'Percentage of match during dates', labels= {'x': 'Match'})
fig_match_perc.update(layout=dict(title=dict(x=0.5)))
fig_match_perc.update_layout(yaxis_title= 'Percentage', width = 800)
fig_match_perc.show()

Finaly, we can see that the objective of this study (determine the reason for a match), concern only a small percentage (16.47%) of the outcome of the date, which justify the need for further analysis.

<h1> 4. Data Analysis, how to get a second date ?  <h1>

<h4> 4.1. Real importance of differents features <h4>

First we are checking the real importance of the features in a match, by looking the percentage of match for each rating for each features

In [18]:


fig_feat_match = make_subplots(rows = 6, cols = 2, subplot_titles = ("Male Features", "Female Features"))

fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['sinc']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['sinc']).sum()['match'].sort_index().values)/dataset.loc[dataset['gender'] == 1,:]['sinc'].value_counts().sort_index().values*100, line=dict(color='#ff7f0e')),
                        row = 1,
                        col = 1,
                        )
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['attr']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['attr']).sum()['match'].values)/dataset.loc[dataset['gender'] == 1,:]['attr'].value_counts().sort_index().values*100, line=dict(color='#ff7f0e')),
                        row = 2,
                        col = 1)
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['intel']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['intel']).sum()['match'].values)/dataset.loc[dataset['gender'] == 1,:]['intel'].value_counts().sort_index().values*100, line=dict(color='#ff7f0e')),
                        row = 3,
                        col = 1)
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['fun']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['fun']).sum()['match'].values)/dataset.loc[dataset['gender'] == 1,:]['fun'].value_counts().sort_index().values*100, line=dict(color='#ff7f0e')),
                        row = 4,
                        col = 1)
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['amb']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['amb']).sum()['match'].values)/dataset.loc[dataset['gender'] == 1,:]['amb'].value_counts().sort_index().values*100, line=dict(color='#ff7f0e')),
                        row = 5,
                        col = 1)
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['shar']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['shar']).sum()['match'].values)/dataset.loc[dataset['gender'] == 1,:]['shar'].value_counts().sort_index().values*100, line=dict(color='#ff7f0e')),
                        row = 6,
                        col = 1)

fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['sinc']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['sinc']).sum()['match'].sort_index().values)/dataset.loc[dataset['gender'] == 0,:]['sinc'].value_counts().sort_index().values*100, line=dict(color='#1f77b4')),
                        row = 1,
                        col = 2,
                        )
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['attr']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['attr']).sum()['match'].values)/dataset.loc[dataset['gender'] == 0,:]['attr'].value_counts().sort_index().values*100, line=dict(color='#1f77b4')),
                        row = 2,
                        col = 2)
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['intel']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['intel']).sum()['match'].values)/dataset.loc[dataset['gender'] == 0,:]['intel'].value_counts().sort_index().values*100, line=dict(color='#1f77b4')),
                        row = 3,
                        col = 2)
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['fun']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['fun']).sum()['match'].values)/dataset.loc[dataset['gender'] == 0,:]['fun'].value_counts().sort_index().values*100, line=dict(color='#1f77b4')),
                        row = 4,
                        col = 2)
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['amb']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['amb']).sum()['match'].values)/dataset.loc[dataset['gender'] == 0,:]['amb'].value_counts().sort_index().values*100, line=dict(color='#1f77b4')),
                        row = 5,
                        col = 2)
fig_feat_match.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['shar']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['shar']).sum()['match'].values)/dataset.loc[dataset['gender'] == 0,:]['shar'].value_counts().sort_index().values*100, line=dict(color='#1f77b4')),
                        row = 6,
                        col = 2)

fig_feat_match.update_layout(
    height=1200, 
    width=1000, 
    yaxis1_title = 'Match percentage',
    yaxis2_title = 'Match percentage',
    yaxis3_title = 'Match percentage',
    yaxis4_title = 'Match percentage',
    yaxis5_title = 'Match percentage',
    yaxis6_title = 'Match percentage',
    yaxis7_title = 'Match percentage',
    yaxis8_title = 'Match percentage',
    yaxis9_title = 'Match percentage',
    yaxis10_title = 'Match percentage',
    yaxis11_title = 'Match percentage',
    yaxis12_title = 'Match percentage',
    xaxis1_title = 'Rated Sincerity',
    xaxis2_title = 'Rated Sincerity',
    xaxis3_title = 'Rated Attractivness',
    xaxis4_title = 'Rated Attractivness',
    xaxis5_title = 'Rated Intelligence',
    xaxis6_title = 'Rated Intelligence',
    xaxis7_title = 'Rated Fun',
    xaxis8_title = 'Rated Fun',
    xaxis9_title = 'Rated Ambition',
    xaxis10_title = 'Rated Ambition',
    xaxis11_title = 'Rated Sharing of interest',
    xaxis12_title = 'Rated Sharing of interest',

    legend_tracegroupgap = 1000,
    showlegend = False,
    title={
        'text': "Match percentage depending on the features rating",
        'y':0.98,
        'x':0.5,
        'xanchor': 'center',
        'yanchor': 'top'},
)

fig_feat_match.show()

/home/triomphant/.local/lib/python3.11/site-packages/plotly/graph_objs/_deprecations.py:378: DeprecationWarning:

plotly.graph_objs.Line is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.scatter.Line
  - plotly.graph_objs.layout.shape.Line
  - etc.


/home/triomphant/.local/lib/python3.11/site-packages/plotly/graph_objs/_deprecations.py:378: DeprecationWarning:

plotly.graph_objs.Line is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.scatter.Line
  - plotly.graph_objs.layout.shape.Line
  - etc.


/home/triomphant/.local/lib/python3.11/site-packages/plotly/graph_objs/_deprecations.py:378: DeprecationWarning:

plotly.graph_objs.Line is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.scatter.Line
  - plotly.graph_objs.layout.shape.Line
  - etc.


/home/triomphant/.local/lib/python3.11/site-packages/plotly/graph_objs/_deprecations.py:378: D

As we assumed looking at the goals the attractiveness is one of the most important attribute for both gender which is coherent with the statement for the men, but contrast with the stated preference for the women,
indeed sincerity and intelligence seems less important than declared compared to physical appearance, as the highest score on attractivness (10/10) get a match alsmot 43% of the time versus 28% of the time for sincerity.

However, suprisingly the most important attributes for both gender seems to be more personality based, as a high sharing interest got the highest match percentage (38% for men, and 53% for women), and the fun seems equivaly imporant that the attractiveness. 

We can also notice that being perceive as too much (10/10) intelligent does not seems to bring more match. The same effect is noticeable for the ambition in women.

<h4> 4.2. How does your opinion on yourself affect your match ? <h4>

In [19]:
fig = make_subplots(rows = 6, cols = 2, subplot_titles = ("Male own rated features", "Female own rated features"))

fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['sinc3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['sinc3_1']).sum()['match'].sort_index().values)/dataset.loc[dataset['gender'] == 1,:]['sinc3_1'].value_counts().sort_index().values*100, 
                        line=dict(color='#ff7f0e')),
                        row = 1,
                        col = 1,
                        )
fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['attr3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['attr3_1']).sum()['match'].values)/dataset.loc[dataset['gender'] == 1,:]['attr3_1'].value_counts().sort_index().values*100, 
                        line=dict(color='#ff7f0e')),
                        row = 2,
                        col = 1)
fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['intel3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['intel3_1']).sum()['match'].values)/dataset.loc[dataset['gender'] == 1,:]['intel3_1'].value_counts().sort_index().values*100, 
                        line=dict(color='#ff7f0e')),
                        row = 3,
                        col = 1)
fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['fun3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['fun3_1']).sum()['match'].values)/dataset.loc[dataset['gender'] == 1,:]['fun3_1'].value_counts().sort_index().values*100, 
                        line=dict(color='#ff7f0e')),
                        row = 4,
                        col = 1)
fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 1,:].groupby(['amb3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 1,:].groupby(['amb3_1']).sum()['match'].values)/dataset.loc[dataset['gender'] == 1,:]['amb3_1'].value_counts().sort_index().values*100, 
                        line=dict(color='#ff7f0e')),
                        row = 5,
                        col = 1)

fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['sinc3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['sinc3_1']).sum()['match'].sort_index().values)/dataset.loc[dataset['gender'] == 0,:]['sinc3_1'].value_counts().sort_index().values*100, 
                        line=dict(color='#1f77b4')),
                        row = 1,
                        col = 2,
                        )
fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['attr3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['attr3_1']).sum()['match'].values)/dataset.loc[dataset['gender'] == 0,:]['attr3_1'].value_counts().sort_index().values*100, 
                        line=dict(color='#1f77b4')),
                        row = 2,
                        col = 2)
fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['intel3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['intel3_1']).sum()['match'].values)/dataset.loc[dataset['gender'] == 0,:]['intel3_1'].value_counts().sort_index().values*100, 
                        line=dict(color='#1f77b4')),
                        row = 3,
                        col = 2)
fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['fun3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['fun3_1']).sum()['match'].values)/dataset.loc[dataset['gender'] == 0,:]['fun3_1'].value_counts().sort_index().values*100,
                        line=dict(color='#1f77b4')),
                        row = 4,
                        col = 2)
fig.add_trace(go.Line(x = dataset.loc[dataset['gender'] == 0,:].groupby(['amb3_1']).sum().index, 
                        y = (dataset.loc[dataset['gender'] == 0,:].groupby(['amb3_1']).sum()['match'].values)/dataset.loc[dataset['gender'] == 0,:]['amb3_1'].value_counts().sort_index().values*100,
                        line=dict(color='#1f77b4')),
                        row = 5,
                        col = 2)

fig.update_layout(
    height=900, 
    width=1000,  
    yaxis1_title = 'Match percentage',
    yaxis2_title = 'Match percentage',
    yaxis3_title = 'Match percentage',
    yaxis4_title = 'Match percentage',
    yaxis5_title = 'Match percentage',
    yaxis6_title = 'Match percentage',
    yaxis7_title = 'Match percentage',
    yaxis8_title = 'Match percentage',
    yaxis9_title = 'Match percentage',
    yaxis10_title = 'Match percentage',
    yaxis11_title = 'Match percentage',
    yaxis12_title = 'Match percentage',
    xaxis1_title = 'Rated Sincerity',
    xaxis2_title = 'Rated Sincerity',
    xaxis3_title = 'Rated Attractivness',
    xaxis4_title = 'Rated Attractivness',
    xaxis5_title = 'Rated Intelligence',
    xaxis6_title = 'Rated Intelligence',
    xaxis7_title = 'Rated Fun',
    xaxis8_title = 'Rated Fun',
    xaxis9_title = 'Rated Ambition',
    xaxis10_title = 'Rated Ambition',
    xaxis11_title = 'Rated Sharing of interest',
    xaxis12_title = 'Rated Sharing of interest',
    title={
        'text': "Match percentage depending on the own features rating",
        'y':0.98,
        'x':0.5,
        'xanchor': 'center',
        'yanchor': 'top'},
    showlegend = False,
    margin=go.layout.Margin(l=80,
    r=50,
    b=10,
    t=80,
    pad = 4
    ),
)

fig.show()

/home/triomphant/.local/lib/python3.11/site-packages/plotly/graph_objs/_deprecations.py:378: DeprecationWarning:

plotly.graph_objs.Line is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.scatter.Line
  - plotly.graph_objs.layout.shape.Line
  - etc.


/home/triomphant/.local/lib/python3.11/site-packages/plotly/graph_objs/_deprecations.py:378: DeprecationWarning:

plotly.graph_objs.Line is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.scatter.Line
  - plotly.graph_objs.layout.shape.Line
  - etc.


/home/triomphant/.local/lib/python3.11/site-packages/plotly/graph_objs/_deprecations.py:378: DeprecationWarning:

plotly.graph_objs.Line is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.scatter.Line
  - plotly.graph_objs.layout.shape.Line
  - etc.


/home/triomphant/.local/lib/python3.11/site-packages/plotly/graph_objs/_deprecations.py:378: D

It seems that having a good opinion of yourself brings more match for men, the trend is harder to figure out for women. However, we could logically think that, men that rate themselves higher got more match,
because they have indeed higher rated attributes. That's why we cannot conclude on these figures and need to compare the self rating with the rating by the date.

<h4> 4.3 What is the importance of confidence ? <h4>

To do this we create some kind of confidence index, which is the difference between the own rating and the date rating (divide by the number of attributes rated)

In [20]:
dataset ['confidence'] = ((dataset['attr3_1'] + dataset['sinc3_1'] + dataset['fun3_1'] + dataset['intel3_1'] + dataset['amb3_1']) - (dataset['attr'] + dataset['sinc'] + dataset['fun'] + dataset['intel'] + dataset['amb']))/5

fig_confidence = make_subplots(rows = 1, cols = 2, subplot_titles = ("Male preferred confidence", "Female preferred confidence"))

fig_confidence.add_trace(go.Scatter(x = dataset.loc[dataset['gender'] == 1,:].groupby(['confidence']).sum().index, 
                                    y =(dataset.loc[dataset['gender'] == 1,:].groupby(['confidence']).sum()['match'].sort_index().values)/dataset.loc[dataset['gender'] == 1,:]['confidence'].value_counts().sort_index().values*100 ,
                                    line=dict(color='#ff7f0e')))
fig_confidence.add_trace(go.Scatter(x = dataset.loc[dataset['gender'] == 0,:].groupby(['confidence']).sum().index, 
                                    y =(dataset.loc[dataset['gender'] == 0,:].groupby(['confidence']).sum()['match'].sort_index().values)/dataset.loc[dataset['gender'] == 0,:]['confidence'].value_counts().sort_index().values*100 ,
                                    line=dict(color='#1f77b4')),
                                    row = 1,
                                    col = 2)
    
fig_confidence.update_layout(
    yaxis1_title = 'Match percentage',
    yaxis2_title = 'Match percentage',
    xaxis1_title = 'Confidence index',
    xaxis2_title = 'Confidence index',
    showlegend = False,
    title={
        'text': "Match percentage depending on the confidence index",
        'y':0.9,
        'x':0.5,
        'xanchor': 'center',
        'yanchor': 'top'})

As we can see on these graphs for both men and women underestimate yourself (or being humble?) seems to bring more match, with 100% of match when the candidates underestimates themselves of approximatly 3 points.

<h4> 4.4. Does the field of study influence the match ? <h4>

As the candidates are mostly students, it could be intresting to see if some study field are considered as more attractive.

In [21]:


fig_study_male = px.bar(dataset, x =dataset.loc[dataset['gender'] == 1,:].groupby(['field']).sum().index, y = (dataset.loc[dataset['gender'] == 1,:].groupby(['field']).sum()['match'].sort_index().values)/dataset.loc[dataset['gender'] == 1,:]['field'].value_counts().sort_index().values*100, color_discrete_sequence =['#1f77b4']*len(dataset)  )
fig_study_female = px.bar(dataset, x =dataset.loc[dataset['gender'] == 0,:].groupby(['field']).sum().index, y = (dataset.loc[dataset['gender'] == 0,:].groupby(['field']).sum()['match'].sort_index().values)/dataset.loc[dataset['gender'] == 0,:]['field'].value_counts().sort_index().values*100, color_discrete_sequence =['#ff7f0e']*len(dataset)  )

figure1_traces = []
figure2_traces = []
for trace in range(len(fig_study_female["data"])):
    figure1_traces.append(fig_study_female["data"][trace])
for trace in range(len(fig_study_male["data"])):
    figure2_traces.append(fig_study_male["data"][trace])

this_figure = make_subplots(rows=2, cols=1, vertical_spacing = 0.35, subplot_titles= ('Male', 'Female')) 

for traces in figure1_traces:
    this_figure.append_trace(traces, row=1, col=1)
for traces in figure2_traces:
    this_figure.append_trace(traces, row=2, col=1)

this_figure.update_layout(height = 1400, 
                        yaxis1_title = 'Match percentage',
                        yaxis2_title = 'Match percentage',
                        margin=go.layout.Margin(l=50,
                                                r=50,
                                                b=100,
                                                t=80,
                                                pad = 4
                                                ),
                        title=dict(text='Percentage of match according to the field studied ',
                                    x=0.5,
                                    y=0.99,
                                    xanchor='center',
                                    yanchor='top'))

As we can see on these graph some of the field of study seems to be more attractive than others.

For men the top 3: 'Master in Public Administration' , 'Sociology', and  'Polish'

For women the top 3: 'Art History/Medecine' , 'Fundraising Management', and 'Money'

However we might need more data for this to be representative, for exemple with the sociology field:

In [22]:
print('Number of candidate sutdying sociology: {}'.format(dataset.loc[(dataset['gender'] == 0) & (dataset['field'] == 'Sociology') ,:]['iid'].nunique()))
print('Average attractivity rating of this candidate: {}'.format(dataset.loc[(dataset['iid'] == 112), 'attr'].mean()))
print('Average attracitivity rating: {}'.format(round(dataset.loc[:, 'attr'].mean(),1)))

Number of candidate sutdying sociology: 1
Average attractivity rating of this candidate: 6.4
Average attracitivity rating: 6.2


We can see that only 1 candidate is studying sociology, and he is for exemple more attractive than the average, so it is hard to make conclusion on the correlation between study field and match

<h1> Conclusion <h1>

<h4> Data quality <h4>

Before conclude on the question 'How to get a second date ?', it is important to make a disclamer about the data, and the possible bias. 

First the data, present a lot of missing values, which can lead to misinterpretation, or make some parameters difficult to evaluate. Then the dataset is still quite small (8378 rows), we don't have enough data to make solid conclusion on some of the parameters because some categories are underrepresented, like the sociology study field exemple seen above.

Finally, the experimental context could include bias compare to what could happend in real life, so trying to get the same kind of data but out of and 'experimental speed dating events' could be intresting. The very short time (4 minutes) dates could also induces bias as it can be difficult to evaluate things like sincerity or ambition on such a little time.

<h4> How to get a second date? <h4>

On the conclusion on how to get a second date we find some 'evident' conclusions as an high physical attractivness. But also more personality based attributes and possible more long-term relationship which contrast with the stated goals of the participants, like:
- shared intrests
- fun

And also more more subtle qualities such as humility ! 

Finally, we get some surprising results like:
- too much of intelligence and ambition (for men), is not optimal in order to get a match.
- The stated most important attributes may significatively differ from the real important attributes, especially for women with the sincerity not being as much important as stated, the explanation could be into some sort of societal norm/pressure, but it is another subject.